### ISSUE TITLE
False positive upon subscription on an instance of a generic type alias

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#2155]
## Environment data

-   Language Server version: 2021.12.1 | 2021.12.2-pre.1
-   OS and version: Windows 7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.12 Anaconda

## Sample code

```python
from __future__ import annotations

from typing import Any, Generic, TypeVar

T = TypeVar("T")

class A(Generic[T]):
    def __getitem__(self, key: Any) -> Any: ...

B = A[T]

b: B[Any] = globals()[""]
item = b[0]
```

## Expected behavior

No error.

## Actual behavior

Got "Expected class type but received ..." on the last line.

**Comment by Azureblade3808** [comment]
This is causing the most important part of the `numpy.typing` to fail to work.

```python
from __future__ import annotations

# Using `numpy` with version above 1.20.0.
import numpy
import numpy.typing

array: numpy.typing.NDArray[numpy.float64] = globals()[""]
value = array[0]
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2021.12.1, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2021121-9-december-2021)

**Comment by Azureblade3808** [comment]
Doesn't seem to get fixed in 2021.12.1 | 2021.12.2-pre.1.

**Comment by Azureblade3808** [comment]
> This issue has been fixed in version 2021.12.1, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2021121-9-december-2021)

Is the fix not published yet? For this issue was found in 2021.12.1 and I haven't see a new version of 2021.12.2 in case there's a typo.

**Comment by erictraut** [comment]
This was fixed after Pylance 2021.12.1 was released.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2021.12.2, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2021122-13-december-2021)
